capstone/src/test/java/com/google/sps/clubs/ClubServletTest.java

```diff
@@ -65,70 +69,114 @@ public void setUp() throws IOException {
     Query query =
         new Query("Club")
             .setFilter(
                 new FilterPredicate(
-                    Constants.CLUB_NAME_PROP,
+                    Constants.PROPERTY_NAME,
                     FilterOperator.EQUAL,
-                    request.getParameter(Constants.CLUB_NAME_PROP)));
+                    request.getParameter(Constants.PROPERTY_NAME)));
     Entity clubEntity = datastore.prepare(query).asSingleEntity();
     Assert.assertEquals(SAMPLE_CLUB_DESC_1, clubEntity.getProperty(Constants.DESCRIP_PROP));
   }
+
+  @Test
+  public void doGet_clubExists() throws ServletException, IOException {
+    ImmutableList<String> expectedMembers =
+        ImmutableList.of("student@example.com", "officer@example.com");
+    ImmutableList<String> expectedOfficers = ImmutableList.of("officer@example.com");
+    ImmutableList<String> expectedAnnouncements = ImmutableList.of("an announcement");
+
+    Entity clubEnt = new Entity("Club");
+    clubEnt.setProperty(Constants.PROPERTY_NAME, SAMPLE_CLUB_NAME);
+    clubEnt.setProperty(Constants.DESCRIP_PROP, "test description");
+    clubEnt.setProperty(Constants.MEMBER_PROP, expectedMembers);
+    clubEnt.setProperty(Constants.OFFICER_PROP, expectedOfficers);
+    clubEnt.setProperty(Constants.WEBSITE_PROP, "website.com");
+    clubEnt.setProperty(Constants.ANNOUNCE_PROP, expectedAnnouncements);
+    datastore.put(clubEnt);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(printWriter);
+    clubServlet.doGet(request, response);
+
+    String responseStr = stringWriter.toString().trim();
+    JsonElement responseJsonElement = new JsonParser().parse(responseStr);
+    JsonObject response = responseJsonElement.getAsJsonObject();
+
+    ImmutableList<String> actualMembers = convertJsonList(response.get(Constants.MEMBER_PROP));
+    ImmutableList<String> actualOfficers = convertJsonList(response.get(Constants.OFFICER_PROP));
+    ImmutableList<String> acutalAnnouncements =
+        convertJsonList(response.get(Constants.ANNOUNCE_PROP));
+
+    Assert.assertEquals(SAMPLE_CLUB_NAME, response.get(Constants.PROPERTY_NAME).getAsString());
+    Assert.assertEquals("test description", response.get(Constants.DESCRIP_PROP).getAsString());
+    Assert.assertEquals(expectedMembers, actualMembers);
+    Assert.assertEquals(expectedOfficers, actualOfficers);
+    Assert.assertEquals("website.com", response.get(Constants.WEBSITE_PROP).getAsString());
+    Assert.assertEquals(expectedAnnouncements, acutalAnnouncements);
+  }
+
+  private ImmutableList<String> convertJsonList(JsonElement responseProp) {
+    ImmutableList<String> converted =
+        Streams.stream(responseProp.getAsJsonArray())
+            .map(element -> element.toString().replaceAll("\"", ""))
+            .collect(toImmutableList());
+    return converted;
+  }
```

Comment:
nit: it's preferred to group tests together and have non-test methods (helpers) at the end